src/ImageSharp/Processing/Dithering/ErrorDiffusion/ErrorDiffuserBase.cs

```diff
@@ -74,9 +74,21 @@ public void Dither<TPixel>(ImageFrame<TPixel> image, TPixel source, TPixel trans
         {
             image[x, y] = transformed;
 
+            // Equal? Break out as there's nothing to pass.
```

Comment:
There are still many optimization opportunities left here, but we need to change (extend?) the `IErrorDiffuser` API for that.

Not related to this PR but `[MethodImpl(MethodImplOptions.AggressiveInlining)]` is unnecessary on `Dither<TPixel>()`. The compiler won't be able to inline it.

Reply:
I thought there were improvements to what can be inlined that would allow this. I could be wrong though. 🤷‍♂️

Reply:
If you invoke a method through interface or virtual call on a base class, it can't be inlined, because the code is not known at JIT time. (The virtual invocation is responsible to dispatch the method at runtime.)